packages/webview_flutter/webview_flutter_wkwebview/lib/src/webkit_webview_controller.dart

```diff
@@ -824,6 +824,54 @@ class WebKitNavigationDelegate extends PlatformNavigationDelegate {
           );
         }
       },
+      didReceiveAuthenticationChallenge: (
+        WKWebView webView,
+        NSUrlAuthenticationChallenge challenge,
+        void Function(
+          NSUrlSessionAuthChallengeDisposition disposition,
+          NSUrlCredential? credential,
+        ) completionHandler,
+      ) {
+        if (challenge.protectionSpace.authenticationMethod ==
```

Comment:
Why does this only support `HTTPBasic`? I'm not that familiar with authentication, but would other methods not work with the response to the completionHandler?

Reply:
I am also not too familiar with the other authentication methods. Instead of making a mistake here, my line of thinking was to take the conservative approach and limit functionality here. In the future, other authentication methods could be added.